src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from "react";
+import { render, fireEvent, waitFor, screen } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+import useMeasureServiceApi, {
+  MeasureServiceApi,
+} from "../../../../api/useMeasureServiceApi";
+import { Measure } from "@madie/madie-models";
+import QDMReporting from "./QDMReporting";
+
+const mockHistoryPush = jest.fn();
+
+jest.mock("react-router-dom", () => ({
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
+
+jest.mock("../../../../api/useMeasureServiceApi");
+const measure = {
+  id: "test measure",
+  measureName: "the measure for testing",
+  cqlLibraryName: "TestCqlLibraryName",
+  ecqmTitle: "ecqmTitle",
+  measurementPeriodStart: "01/01/2022",
+  measurementPeriodEnd: "12/02/2022",
+  createdBy: "john doe",
+  measureSetId: "testMeasureId",
+  acls: [{ userId: "othertestuser@example.com", roles: ["SHARED_WITH"] }], //#nosec
+} as unknown as Measure;
+
+jest.mock("@madie/madie-editor", () => ({
+  synchingEditorCqlContent: jest.fn().mockResolvedValue("modified cql"),
+  parseContent: jest.fn(() => []),
+  validateContent: jest.fn().mockResolvedValue({
+    errors: [],
+    translation: { library: "NewLibName" },
+  }),
+}));
+
+jest.mock("@madie/madie-util", () => ({
+  useOktaTokens: jest.fn(() => ({
+    getAccessToken: () => "test.jwt",
+  })),
+  useKeyPress: jest.fn(() => false),
+  measureStore: {
+    updateMeasure: jest.fn(),
+    state: measure,
+    initialState: jest.fn(),
+    subscribe: () => {
+      return { unsubscribe: () => null };
+    },
+  },
+  routeHandlerStore: {
+    subscribe: (set) => {
+      set();
+      return { unsubscribe: () => null };
+    },
+    updateRouteHandlerState: () => null,
+    state: { canTravel: true, pendingPath: "" },
+    initialState: { canTravel: true, pendingPath: "" },
+  },
+
+  checkUserCanEdit: jest.fn(() => {
+    return true;
+  }),
+}));
+
+const useMeasureServiceApiMock =
+  useMeasureServiceApi as jest.Mock<MeasureServiceApi>;
+
+let serviceApiMock: MeasureServiceApi;
+
+describe("QDMReporting component", () => {
+  const { getByTestId, findByTestId, getByText } = screen;
+
+  test("QDMReporting renders to correctly with defaults", async () => {
+    render(<QDMReporting />);
+
+    expect(getByTestId("rateAggregationText")).toHaveValue("");
+  });
+
+  test("Change enables Discard button and click Discard resets the form", async () => {
+    render(<QDMReporting />);
+
+    const rateAggregation = getByTestId(
+      "rateAggregationText"
+    ) as HTMLInputElement;
+
+    fireEvent.change(rateAggregation, {
+      target: { value: "Test" },
+    });
+    expect(rateAggregation.value).toBe("Test");
+
+    const cancelButton = getByTestId("cancel-button");
+    expect(cancelButton).toBeInTheDocument();
+    await waitFor(() => expect(cancelButton).toBeEnabled());
+    act(() => {
+      fireEvent.click(cancelButton);
```

Comment:
FireEvent is already wrapped in act, So we don;t have to explicitly wrap it in act

Reply:
done